regex_src/regexMain.c

```diff
@@ -104,18 +105,32 @@ struct test_case {
 	const regex_char_t *string;	/* NULL : end of tests. */
 };
 
-static void run_tests(struct test_case* test, int verbose, int silent)
+static int run_tests(struct test_case* test, int verbose, int silent)
 {
 	int error;
 	const regex_char_t *ptr;
 	struct regex_machine* machine = NULL;
 	struct regex_match* match;
+	const char *platform_name;
 	int begin, end, id, finished;
 	int success = 0, fail = 0;
 
 	if (!verbose && !silent)
 		printf("Pass -v to enable verbose, -s to disable this hint.\n\n");
 
+	platform_name = regex_get_platform_name();
+
+	/* sljit_get_platform_namei() is meant to return a string literal so this should never happen, but... */
+	if (platform_name == NULL) {
+		printf("ABORT: sljit_get_platform_name() broken\n");
+		return 1;
+	}
+
+	if (strlen(platform_name) >= PLATFORM_NAME_BUFFERSIZE_LIMIT) {
```

Comment:
This is a char*, why should it have a limit?

Reply:
> We've had to bump up the documented limit twice now

If you are referring to the recent preemptive bump I did before the release of 10.47 and which was sadly not large enough; that was actually a revert from some documentation change done before in f27ca13 (Update pcre2_pattern_info to return a length if "where" is NULL., 2014-10-16).

The fact that the standalone jit test source code uses 128 code units, seems to contradict as that being the lesson.

Reply:
Hmm, maybe I'm misremembering. You bumped the documented size requirement for PCRE2_CONFIG_JITTARGET before 10.4; Philip bumped the size requirement for PCRE2_CONFIG_VERSION back in 2017 (10.30). Maybe there haven't been any other increases, and it just feels like "deja vu".

Either way - I don't want us to document current implementation internals like "71 bytes is the required size". That will never last. We should update the documentation to make it clear that it's not safe to pass in a fixed buffer and assume the string will fit.

Reply:
> I would potentially be willing to make another version of pcre2_config with an extra argument... but the benefit is very minor.

I was thinking it will be better to create a new API specially designed to extract these "strings" and make sure that it includes a parameter for buffersize, maybe `pcre2_strlcpy()`? and eventually deprecate the `pcre2_config()` that use a `void *` to return strings in favor of it.

Reply:
Exactly, a `pcre2_config_str(int what, PCRE2_UCHAR *buf, PCRE2_SIZE bufsize)` function.

But it's not strictly necessary, since the current API does work, if callers just call it twice, passing NULL to check they have a big enough buffer.